app/src/main/java/org/mozilla/fenix/settings/address/view/AddressEditorView.kt

```diff
@@ -36,26 +43,64 @@ class AddressEditorView(
         binding.saveButton.setOnClickListener {
             saveAddress()
         }
+
+        address?.let { address ->
+            binding.emailInput.setText(address.email)
+            binding.phoneInput.setText(address.tel)
+
+            binding.firstNameInput.setText(address.givenName)
+            binding.middleNameInput.setText(address.additionalName)
+            binding.lastNameInput.setText(address.familyName)
+
+            binding.streetAddressInput.setText(address.streetAddress)
+            binding.cityInput.setText(address.addressLevel2)
+            binding.stateInput.setText(address.addressLevel1)
+            binding.zipInput.setText(address.postalCode)
+
+            binding.deleteButton.apply {
+                isVisible = true
+                setOnClickListener { view ->
+                    showConfirmDeleteAddressDialog(view.context, address.guid)
+                }
+            }
+        }
     }
 
     internal fun saveAddress() {
         binding.root.hideKeyboard()
 
-        interactor.onSaveAddress(
-            UpdatableAddressFields(
-                givenName = binding.firstNameInput.text.toString(),
-                additionalName = binding.middleNameInput.text.toString(),
-                familyName = binding.lastNameInput.text.toString(),
-                organization = "",
-                streetAddress = binding.streetAddressInput.text.toString(),
-                addressLevel3 = "",
-                addressLevel2 = "",
-                addressLevel1 = "",
-                postalCode = binding.zipInput.text.toString(),
-                country = "",
-                tel = binding.phoneInput.text.toString(),
-                email = binding.emailInput.text.toString()
-            )
+        val addressFields = UpdatableAddressFields(
+            givenName = binding.firstNameInput.text.toString(),
+            additionalName = binding.middleNameInput.text.toString(),
+            familyName = binding.lastNameInput.text.toString(),
+            organization = "",
+            streetAddress = binding.streetAddressInput.text.toString(),
+            addressLevel3 = "",
+            addressLevel2 = "",
+            addressLevel1 = "",
+            postalCode = binding.zipInput.text.toString(),
+            country = "",
+            tel = binding.phoneInput.text.toString(),
+            email = binding.emailInput.text.toString()
         )
+
+        if (address != null) {
+            interactor.onUpdateAddress(address.guid, addressFields)
+        } else {
+            interactor.onSaveAddress(addressFields)
+        }
+    }
+
+    internal fun showConfirmDeleteAddressDialog(context: Context, guid: String) {
```

Comment:
Not quite sure what to think about this. Instinctively, I feel like this should be called through the usual interactor/controller, but I don't know if I have a strong opinion about it.

Reply:
I see the pair interactor/controller usage as a bridge between the actions in the interface and the store and sometimes state. Showing a UI piece (a confirmation dialogue in this case) should be the responsibility of the fragment/view IMO. This would also allow for additional testing of the dialog behavior in the future.

Reply:
I generally agree with your stance. Where I would say I differ or don't hold a strong opinion on is the exact scope of what should be handled in the controller - I think handling all `Store` interactions in there makes sense, but otherwise, I would be looking to extract as much functionality out of the Fragment to make functionality easier to tests. Otherwise, I would say the Interactor is called for all user interactions, which is why I am on the fence since this is technically called through a user tapping on the "Delete button", but it's also called through the menu options.

The View in this case does satisfy the fact that we aren't putting the dialog responsibility in the fragment and thus making it easier to test. So, I could've gone either direction.